quickstart.rb

```diff
@@ -45,26 +43,68 @@ def authorize
 apiCall = Google::Apis::DriveV3::DriveService.new
 apiCall.client_options.application_name = APPLICATION_NAME
 apiCall.authorization = authorize
-# List the 1000 most recently modified files.
+# List the most recently modified GIF files.
 response = apiCall.list_files(page_size: 10,
                               fields: 'nextPageToken, files(id, name, kind, mime_type)',
                               q: "mimeType='image/gif'",
                               spaces: 'photos'
                               )
 
 
-
+# Display output
 puts 'Files:'
 puts 'No files found' if response.files.empty?
+
 response.files.each do |file|
-  #puts file.inspect
-  #puts
-  puts "#{file.name} (#{file.id}) (#{file.kind}) ((#{file.mime_type}))"
 
-  puts "================="
+  #puts file.inspect
+  #puts "#{file.name} (#{file.id}) (#{file.mime_type})"
+  puts ""
   aFile = apiCall.get_file(file.id,
-                            {fields: 'id, name, kind, mime_type, thumbnailLink, webContentLink, webViewLink'}
+                            {fields: 'id, name, mimeType, thumbnailLink, webViewLink, webContentLink, createdTime, ownedByMe, size, imageMediaMetadata'}
                           )
-  puts aFile.inspect
-  puts "==========="
+
+
+
+  puts "File: #{aFile.name}"
+  puts "ID: #{aFile.id}"
+  puts "Created: #{aFile.created_time}"
+  puts "View: #{aFile.web_view_link}"
+  puts "Download: #{aFile.web_content_link}"
+  puts "Thumbnail #{aFile.thumbnail_link}"
+  puts "Mine: #{aFile.owned_by_me}"
+  puts "Dimensions: #{aFile.image_media_metadata.width}x#{aFile.image_media_metadata.height}"
+  puts "Size: #{aFile.size}bytes"
+  if aFile.image_media_metadata.location != nil
+    puts "Loc: #{aFile.image_media_metadata.location.latitude}, #{aFile.image_media_metadata.location.longitude}"
+  else
+    puts "Loc: 0, 0"
+  end
+
+  #puts aFile.inspect
 end
+
+# note: I don't think exporting of gifs works. :(
+#exportFile = apiCall.export_file("1hqnXJsVTSgv3H2gsFoyUVKLQTNiaBcOD8Q", "image/gif")
+
+
+
+## TODO
+#1 Save a copy of gif file (maybe via cUrl?) to gifs.travel server using webContentLink
+  # Windows PowerShell example => Invoke-WebRequest 'https://drive.google.com/uc?id=0B2tqn9-EHK-ZU3ZOWmc1dHR6OFU&export=download' -OutFile test-ANIMATION.gif
+#3 Write to an RSS file (XML) the properties of each new gif
+  # read existing RSS file off server
+  # parse RSS to get pubDate from channel > item[0]
+  # convert pubDate to date
+  # loop through 1000 most recent gifs with Drive.API call {note: that initially need to run through ALL results (not just top 1000)}
+  # if gif.created_time > pubDate then add new item node to rss for each new gif with all metadata
```

Comment:
I don't understand a word. I would store the gifs data from Google on a data base...

Reply:
Do you think a database would be overkill? I'm not sure what advantage would be gained by using a database over using the XML/RSS file. Curious to hear your thoughts.